ironic/drivers/modules/drac/management.py

```diff
@@ -302,14 +314,170 @@ def set_boot_device(node, device, persistent=False):
+                    {'node':task.node.uuid})
+            drac_utils.wait_for_host(redfish_hostname)
+            time.sleep(ready_wait_time)
+            drac_utils.wait_until_idrac_is_ready(oem_manager,
+                        redfish_hostname,
+                        IDRAC_IS_READY_RETRIES,
+                        IDRAC_IS_READY_RETRY_DELAY_SEC)
+        else:
+            error_msg = ("Failed to reset idrac for %(node)s" %
+                        {'node': task.node.uuid})
+            LOG.error(error_msg)
+            raise exception.RedfishError(error=error_msg)
 
+        return None
+
+    @METRICS.timer('DracRedfishManagement.known_good_state')
+    @base.clean_step(priority=0)
+    def known_good_state(self, task):
+        """Reset the iDRAC, Clear the job queue.
+
+        :param task: a TaskManager instance containing the node to act on.
+        :returns: None if it is completed.
+        :raises: RedfishError on an error.
+        """
+        self.reset_idrac(task)
+        self.clear_job_queue(task)
+        return None
+
+    @task_manager.require_exclusive_lock
+    def set_boot_device(self, task, device, persistent=False):
+        if task.node.driver_internal_info.get("clean_steps"):
+            if task.node.driver_internal_info.get("clean_steps")[0].get(
+                    'step') in _CLEAR_JOBS_CLEAN_STEPS:
+                system = redfish_utils.get_system(task.node)
+                for manager in system.managers:
+                    try:
+
+                        oem_manager = manager.get_oem_extension('Dell')
+                    except sushy.exceptions.OEMExtensionNotFoundError as e:
+                        error_msg = (_("Search for Sushy OEM extension package"
+                                "'sushy-oem-idrac' failed for node %(node)s."
+                                "Ensure it is installed. Error: %(error)s") %
+                                {'node': task.node.uuid, 'error': e})
+                        LOG.error(error_msg)
+                        raise exception.RedfishError(error=error_msg)
+
+                    try:
+
+                        unfinished_jobs = oem_manager.get_unfinished_jobs(task)
+                        if unfinished_jobs == []:
+                            info_msg = ("Not found any unfinished jobs for node"
+                                        "%(node)s" %
+                                        {'node':task.node.uuid})
+                            LOG.info(info_msg)
+                        else:
+                            delete_job_response = oem_manager.delete_jobs(
+                                                    task,
+                                                    job_ids = unfinished_jobs)
+
+                    except sushy.exceptions.SushyError as e:
+                        error_msg = ("Sushy OEM extension Python package "
+                                    "sushy-oem-idrac failed to clear job queue"
```

Comment:
Can we change the above to "sushy-oem-idrac failed to delete unfinished jobs"

Reply:
Addressed. Added meaningful sentence.